wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 1 - Quick Start

Issue #15189

Based on the Quick Start Phase 1 design, this PR 
- Adds progress indicator 
- Adds a header icon
- Removes row icons
 
in the quick start block. 

To test:

1. Go to My Site -> App Settings -> Test feature configuration.
2. Disable `QuickStartDynamicCardsFeatureConfig`, enable `MySiteImprovementsFeatureConfig`
3. Restart the app
4. Select a site with quick start in progress
5. Notice that the quick start block contains the following updates:
   - ✅  progress indicator is copied from the [dynamic quick start VH](https://github.com/wordpress-mobile/WordPress-Android/blob/9088afbcfe30b71bbb1974eaf728e0cbe802932f/WordPress/src/main/java/org/wordpress/android/ui/mysite/QuickStartCardViewHolder.kt#L61-L64). It gets updated based on the completed tasks count. 
   - ✅  header icon is present before the title
   - ✅  row icons are removed
   - ✅  subtitle is renamed from `$completedCount of $total` to `$completed out of $total` (this will be reflected in the old my site fragment as well on `develop`)

<img src=https://user-images.githubusercontent.com/1405144/129565101-5577f871-349b-403e-b8d9-d8e746fa636c.jpg  width=500>

Note:
This PR is still in draft as I need to fine-tune the UI based on the final design to be provided in Zeplin.

## Regression Notes
1. Potential unintended areas of impact
    - `ImprovedMySiteFragment` -> `Quick Start - Dynamic Card` (`MySiteImprovementsFeatureConfig` - ON, `QuickStartDynamicCardsFeatureConfig` - ON )
    - `MySiteFragment` -> `Quick Start` (`MySiteImprovementsFeatureConfig` - OFF)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the updated UI elements

PR submission checklist:
